src/chainstate/stacks/boot/exit-at-rc.clar

```diff
@@ -0,0 +1,199 @@
+;; The .exit-at-rc contract
+;; Error codes
+(define-constant ERR_AMOUNT_NOT_POSITIVE 2)
+(define-constant ERR_PREVIOUS_VOTE_VALID 7)
+(define-constant ERR_ALREADY_VETOED 9)
+(define-constant ERR_UNAUTHORIZED_CALLER 10)
+(define-constant ERR_VOTER_NOT_STACKING 13)
+(define-constant ERR_BURN_BLOCK_HEIGHT_TOO_LOW 14)
+(define-constant ERR_FETCHING_BLOCK_INFO 16)
+(define-constant ERR_NOT_ALLOWED 19)
+(define-constant ERR_INVALID_PROPOSED_RC 21)
+
+;; Constants
+(define-constant ABSOLUTE_MINIMUM_EXIT_RC u33)
+(define-constant MAXIMUM_RC_BUFFER_FROM_PRESENT u25)
+(define-constant MINIMUM_RC_BUFFER_FROM_PRESENT u6)
+
+;; Data vars
+(define-data-var pox-reward-cycle-length uint POX_REWARD_CYCLE_LENGTH)
```

Comment:
Instead of duplicating this here, would it be better if the contract simply pulled this information from `.pox` via `get-pox-info`?

Reply:
This also applies to `first-burnchain-block-height` below.

Reply:
It would also be more efficient to store all of these vars as a single tuple data variable, since then reading it would incur only one MARF read.  The initial values for these variables could be instantiated directly from `contract-call?`'s to `.pox`.

Reply:
read-only methods are also callable by other contracts, so the initial values for these variables should be settable by doing something like `(contract-call? .pox get-pox-info)`

Reply:
If these are settable via the pox get-pox-info call, then you should be able to switch them to constants (rather than data vars), which would simplify some of the contract code.